diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringTemplateFormatRule.kt

```diff
@@ -41,8 +45,9 @@ class StringTemplateFormatRule(private val configRules: List<RulesConfig>) : Rul
     @Suppress("UnsafeCallOnNullableType")
     private fun handleLongStringTemplate(node: ASTNode) {
         // Checking if in long templates {a.foo()} there are function calls or class toString call
-        if (node.findAllNodesWithSpecificType(COLONCOLON).isEmpty() &&
-                node.findAllNodesWithSpecificType(DOT_QUALIFIED_EXPRESSION).isEmpty()) {
+        if (node.findAllNodesWithSpecificType(COLONCOLON).isEmpty()
+                && node.findAllNodesWithSpecificType(DOT_QUALIFIED_EXPRESSION).isEmpty()
+                && bracesCanBeOmitted(node)) {
```

Comment:
Maybe instead of checking forbidden element types you can specify allowed types? I think, `REFERENCE_EXPRESSION`s might cover most of needed cases.

Reply:
${a} in this case `a` is also considered as `REFERENCE_EXPRESSION`

Reply:
Yes, and that's why it should be fixed